scipy/integrate/tests/banded5x5.pyf

```diff
@@ -0,0 +1,56 @@
+!    -*- f90 -*-
+! Note: the context of this file is case sensitive.
+
+python module _test_odeint_banded ! in 
+    interface  ! in :_test_odeint_banded
+        subroutine getbands(jac) ! in :_test_odeint_banded:banded5x5.f
+            double precision dimension(4,5),intent(out) :: jac
+            double precision dimension(4,5) :: bands
+            common /jac/ bands
+        end subroutine getbands
+        subroutine banded5x5(n,t,y,f) ! in :_test_odeint_banded:banded5x5.f
+            integer, optional,check(len(y)>=n),depend(y) :: n=len(y)
+            double precision :: t
+            double precision dimension(n) :: y
+            double precision dimension(n),depend(n) :: f
+            double precision dimension(4,5) :: bands
+            common /jac/ bands
+        end subroutine banded5x5
+        subroutine banded5x5_jac(n,t,y,ml,mu,jac,nrowpd) ! in :_test_odeint_banded:banded5x5.f
+            integer, optional,check(len(y)>=n),depend(y) :: n=len(y)
+            double precision :: t
+            double precision dimension(n) :: y
+            integer :: ml
+            integer :: mu
+            double precision dimension(nrowpd,n),depend(n) :: jac
+            integer, optional,check(shape(jac,0)==nrowpd),depend(jac) :: nrowpd=shape(jac,0)
+            double precision dimension(4,5) :: bands
+            common /jac/ bands
+        end subroutine banded5x5_jac
+        subroutine banded5x5_bjac(n,t,y,ml,mu,bjac,nrowpd) ! in :_test_odeint_banded:banded5x5.f
+            integer, optional,check(shape(bjac,1)==n),depend(bjac) :: n=shape(bjac,1)
+            double precision :: t
+            double precision dimension(5) :: y
+            integer :: ml
+            integer :: mu
+            double precision dimension(nrowpd,n) :: bjac
+            integer, optional,check(shape(bjac,0)==nrowpd),depend(bjac) :: nrowpd=shape(bjac,0)
+            double precision dimension(4,5) :: bands
+            common /jac/ bands
+        end subroutine banded5x5_bjac
+        subroutine banded5x5_solve(y,nsteps,dt,jt,nst,nfe,nje) ! in :_test_odeint_banded:banded5x5.f
+            double precision dimension(5),intent(inout) :: y
+            integer intent(in) :: nsteps
+            double precision intent(in) :: dt
+            integer intent(in) :: jt
+            integer intent(out) :: nst
+            integer intent(out) :: nfe
+            integer intent(out) :: nje
+            double precision dimension(4,5) :: jacband
+            common /jac/ jacband
+        end subroutine banded5x5_solve
+    end interface 
+end python module _test_odeint_banded
+
+! This file was auto-generated with f2py (version:1.21.0).
```

Comment:
Is this an accidentally-committed auto-generated file or am I missing something?

Reply:
No, that's on purpose. We have signature files (`.pyf`) for most Fortran code, but not for these tests. And somehow there's an `f2py` bug where this worked with the signature file, but not without. And I did check with the `f2py` devs that this is the best option.

Reply:
Just so I understand--this means the tests currently aren't running for us but they will start running after this PR is merged, or they are currently running for us, but it is the result of some buggy behavior?

Reply:
The tests currently do work, but in a way that's not portable to Meson (= feeding only `.f` files to `config.add_extension` which invokes `f2py` from within `numpy.distutils`).

There are two other ways, asking `f2py` to output a `modulexxx.c` file with or without a signature file. The latter was not working, the former does work. This is a bug in `f2py`; the without-a-signature-file case should be equivalent to doing it from within `add_extension`, but it's not.

Reply:
See https://numpy.org/devdocs/f2py/buildtools/index.html#basic-concepts for the different flavors. The `f2py` UX is a bit of a mess.